flutter/lib/benchmark/benchmark.dart

```diff
@@ -84,485 +60,225 @@ class Benchmark {
   }
 }
 
-class MiddleInterface {
-  final List<Benchmark> benchmarks;
-
-  MiddleInterface._(this.benchmarks);
-
-  static Future<MiddleInterface> create(
-      File configFile, BackendInfo backendInfo) async {
-    final tasks = getMLPerfConfig(await configFile.readAsString());
-
-    final benchmarks = <Benchmark>[];
-    for (final task in tasks.task) {
-      for (final model in task.model) {
-        final benchmarkSetting = backendInfo.settings.benchmarkSetting
-            .singleWhereOrNull((setting) => setting.benchmarkId == model.id);
-        if (benchmarkSetting == null) continue;
-
-        benchmarks.add(Benchmark(benchmarkSetting, task, model));
-      }
-    }
-
-    return MiddleInterface._(benchmarks);
-  }
-
-  /// The list of URL or file names to download.
-  List<String> data() {
-    // TODO unify with listSelectedResources
-    final result = <String>[];
-
-    for (final b in benchmarks) {
-      result.add(b.taskConfig.liteDataset.path);
-      result.add(b.taskConfig.liteDataset.groundtruthSrc);
-
-      result.add(b.taskConfig.dataset.path);
-      result.add(b.taskConfig.dataset.groundtruthSrc);
-
-      result.add(b.taskConfig.testDataset.path);
-      result.add(b.taskConfig.testDataset.groundtruthSrc);
-
-      result.add(b.benchmarkSetting.src);
-    }
+class BenchmarkResult {
+  final double score;
+  final String accuracy;
+  final String backendName;
```

Comment:
We get backend name/description string from the function named `mlperf_backend_name`:
https://github.com/mlcommons/mobile_app_open/blob/857fe9cdd6e49528ab3df53861219a7a0ff677e6/android/cpp/c/backend_c.h#L39-L40
This function also has clear documentation that it should return name, not some other info.
I think we should rename all `backendDescription` variables in our code to `backendName` while we use `mlperf_backend_name` function to get this backend description string.

We may need to change naming if/when we use another backend function to get data.
Changing variable names in Flutter code is fast and easy, we don't depend on any vendor to do it, so I think it would be better to reflect the actual data stored in the variable.

Reply:
The code to read the backend name is on the Flutter side
https://github.com/mlcommons/mobile_app_open/blob/857fe9cdd6e49528ab3df53861219a7a0ff677e6/flutter/cpp/flutter/main.cc#L94
So theoretically, we can create the string `backendDescription` from multiple functions (like the not used `mlperf_backend_vendor_name` one)
Currently, the setting screen show `backend name | accelerator name` (before benchmark run) and the result screen show only `backend name`, but there is an issue to also show the `accelerator name`.
https://github.com/mlcommons/mobile_app_open/issues/203#issuecomment-1015102664
So, I prefer to just change the variable name here.

Reply:
I would prefer to use naming appropriate to what the variable is _currently_ used for.
We can change the name in PR for #203. I don't know how much time it will take to specify and approve changes for #203, and then to implement it, and I would prefer to keep variable naming clear until that future PR is merged.

Reply:
Ok. Also, please change `backendDescription` to `backendName` in other places, where it appropriate.

Reply:
Done